src/Controller.ts

```diff
@@ -38,9 +40,24 @@ export class Controller {
             debug ? await this._enableAllBPs() : await this._disableAllBPs();
 
             app.activeSessionName = targetConfig.name;
+            let jmxport = await(getPort());
+            app.jmxPort = jmxport;
+            let vmArgs = '-Dcom.sun.management.jmxremote ' +
+                `-Dcom.sun.management.jmxremote.port=${jmxport} `+ 
+                '-Dcom.sun.management.jmxremote.authenticate=false ' + 
+                '-Dcom.sun.management.jmxremote.ssl=false ' +
+                '-Djava.rmi.server.hostname=localhost '+ 
+                '-Dspring.application.admin.enabled=true';
+            if (targetConfig.vmArgs) {
+                //TODO: smarter merge? What if user is trying to enable jmx themselves on a specific port they choose, for example?
+                vmArgs = vmArgs + ' ' +targetConfig.vmArgs;
+            }
             const ok: boolean = await vscode.debug.startDebugging(
                 vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(app.path)),
-                Object.assign({}, targetConfig, { noDebug: !debug })
+                Object.assign({}, targetConfig, { 
+                    noDebug: !debug,
+                    vmArgs: vmArgs
```

Comment:
vmArgs

Reply:
I don't understand what you want me to change here..

Reply:
I think he want to say:
`vmArgs: vmArgs` can be simply written as `vmArgs` here.